aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/StorageInstrumentationTestBase.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.storage.StorageAccessLevel;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.options.StorageUploadFileOptions;
+import com.amplifyframework.storage.result.StorageUploadFileResult;
+import com.amplifyframework.storage.s3.utils.S3RequestUtils;
+import com.amplifyframework.testutils.AmplifyTestBase;
+import com.amplifyframework.testutils.Await;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.services.s3.AmazonS3Client;
+import org.json.JSONObject;
+import org.junit.BeforeClass;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Abstract test base for Storage instrumented tests. This test contains
+ * basic static methods that involve direct interactions with low-level
+ * {@link AmazonS3Client} and {@link AWSMobileClient}.
+ */
+public abstract class StorageInstrumentationTestBase extends AmplifyTestBase {
```

Comment:
Let's get rid of both the `AmplifyTestBase` and `StorageInstrumentationTestBase`. This is a well-intentioned idea, at first glance -- an attempt to re-use some code. However -- and I've seen this more times than I can count -- here's what tends to happen with this pattern. The base classes grow and grow, and they have all kinds of stuff put into them. Because, all of the things that are being added are to support tests! And that's the name of the base class. But, it becomes very hard to maintain.

So, a better way to organize these things is to create a ton of small utility classes, and then pick and choose them when you're writing your test class(es).

You could split this class out into the following static utilities, perhaps:
```
File tempFile = TempFile.create() // Gets a random name?
File tempPotatoFile = TempFile.create("potato.txt");
File tempPotatoFile30bytes = TempFile.create("potato.txt", 30 /* bytes */);
```

```
S3Assert.exists("bucketName");
```

For the other async APIs, the pattern we have been using is to create a test-only synchronous version, for convenience. You already have a `latchedUploadAndConfirm` method. So, if you rework this into like a :
```
final class SynchronousStorage {
    StorageUploadFileResult uploadFile(StorageOptions options) throws StorageException {
        return Await.result(...);
    }
```

You can have that stuff off in a single-responsibility component.